flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/split/MySqlBinlogSplit.java

```diff
@@ -183,17 +183,29 @@ public static MySqlBinlogSplit appendFinishedSplitInfos(
      *
      * <p>When restore from a checkpoint, the finished split infos may contain some splits from the
      * deleted tables. We need to remove these splits from the total finished split infos and update
-     * the size.
+     * the size, while also removing the outdated tables from the table schemas of binlog split.
      */
     public static MySqlBinlogSplit filterOutdatedSplitInfos(
             MySqlBinlogSplit binlogSplit, Tables.TableFilter currentTableFilter) {
+        Map<TableId, TableChange> newTableSchemas =
```

Comment:
```suggestion
        Map<TableId, TableChange> filteredTableSchemas =
```

Reply:
Thanks for the review, [ruanhang1993](https://github.com/ruanhang1993), I've applied your suggestions.